src/Eloquent/Model.php

```diff
@@ -270,6 +299,24 @@ public function fromJson($value, $asObject = false)
         return Json::decode($value ?? '', ! $asObject);
     }
 
+    /** @inheritdoc */
+    protected function castAttribute($key, $value)
+    {
+        $castType = $this->getCastType($key);
+
+        switch ($castType) {
```

Comment:
The `match` operator would be perfect for this usage.

Reply:
I used `switch` because of its parent method that used `switch` instead of `match`. I'm going to replace this with `match` though.